### ISSUE TITLE
OWNCLOUD_SUB_URL when used, PropFind returns 405: Is files folder empty?

### ISSUE THREAD

**Issue by jariojose** [owncloud-docker/server#383]
Hello, all!

I am using Nginx as Reverse Proxy in a separeted machine ports 80.  The nginx conf:

```
    location /owncloud {
        proxy_pass http://owncloud_ip:owncloud_port/owncloud;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_redirect off;
        proxy_buffering off;
    }
```

OwnCloud PropFind docker log:

` - - [21/Mar/2023:12:42:11 +0000] "PROPFIND /owncloud/remote.php/dav/files/admin/ HTTP/1.0" 405 679 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36"`

Browser Console

![image](https://user-images.githubusercontent.com/6197314/226610434-653dddc3-2c4b-462c-ae21-e7966fd8fd32.png)

Compose setup (I putted 000000 in some fields only to share here):

```
services:
  owncloud:
    image: owncloud/server:latest
    container_name: OwnCloud
    restart: always
    ports:
      - 18080:8080
    environment:
      - OWNCLOUD_DOMAIN=00000000
      - OWNCLOUD_SUB_URL=/owncloud
      - OWNCLOUD_TRUSTED_DOMAINS=000000000
      - OWNCLOUD_DB_TYPE=pgsql
      - OWNCLOUD_DB_NAME=owncloud
      - OWNCLOUD_DB_USERNAME=postgres
      - OWNCLOUD_DB_PASSWORD=00000000
      - OWNCLOUD_DB_HOST=000000
      - OWNCLOUD_ADMIN_USERNAME=admin
      - OWNCLOUD_ADMIN_PASSWORD=aaaa
      - OWNCLOUD_REDIS_ENABLED=true
      - OWNCLOUD_REDIS_HOST=0000000
      - OWNCLOUD_REDIS_PORT=0000000

    volumes:
      - /volume1/Docker/OwnCloud:/mnt/data


networks:
  default:
    name: applications-dev-network
    external: true
```

If I use without SUB_URL, the owncloud works fine.

**Comment by mkuf** [comment]
Same issue here, slightly different Setup w/ traefik.  

The admin panel reports issues with WebDAV.  
![image](https://user-images.githubusercontent.com/11882946/230801020-2c3f8b8d-a60b-4396-8aae-337d8e41cdaf.png)

The files app does not show any directories or files, but they exist on the underlying filesystem.  
Docker logs show a 405 error when a PROPFIND is sent to the container.
```
172.19.0.17 - - [09/Apr/2023:23:17:43 +0000] "GET /owncloud/apps/files/ HTTP/1.1" 200 5830 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
172.19.0.17 - - [09/Apr/2023:23:17:43 +0000] "GET /owncloud/core/js/oc.js?v=cb37f41f070a4da019c1e04eda1e29cb HTTP/1.1" 200 8245 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
172.19.0.17 - - [09/Apr/2023:23:17:43 +0000] "GET /owncloud/core/vendor/jquery/dist/jquery.min.map HTTP/1.1" 302 934 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
172.19.0.17 - - [09/Apr/2023:23:17:43 +0000] "GET /owncloud/apps/files/ HTTP/1.1" 200 5826 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
172.19.0.17 - - [09/Apr/2023:23:17:43 +0000] "GET /owncloud/core/vendor/es6-promise/es6-promise.auto.map HTTP/1.1" 302 934 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
172.19.0.17 - - [09/Apr/2023:23:17:43 +0000] "GET /owncloud/core/vendor/underscore/underscore-umd.js.map HTTP/1.1" 302 934 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
172.19.0.17 - - [09/Apr/2023:23:17:44 +0000] "GET /owncloud/index.php/apps/richdocuments/ajax/settings.php HTTP/1.1" 200 742 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
172.19.0.17 - - [09/Apr/2023:23:17:44 +0000] "GET /owncloud/ocs/v2.php/apps/notifications/api/v1/notifications?format=json HTTP/1.1" 302 665 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
172.19.0.17 - - [09/Apr/2023:23:17:44 +0000] "GET /owncloud/apps/files/ HTTP/1.1" 200 5833 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
172.19.0.17 - - [09/Apr/2023:23:17:44 +0000] "PROPFIND /owncloud/remote.php/dav/files/markus/ HTTP/1.1" 405 660 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
172.19.0.17 - - [09/Apr/2023:23:17:44 +0000] "GET /owncloud/index.php/apps/files/ajax/getstoragestats.php?dir=%2F HTTP/1.1" 200 812 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
172.19.0.17 - - [09/Apr/2023:23:17:44 +0000] "GET /owncloud/apps/files/ HTTP/1.1" 200 5828 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
172.19.0.17 - - [09/Apr/2023:23:17:44 +0000] "GET /owncloud/apps/files/ HTTP/1.1" 200 5568 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
```

from my compose file:
```
  owncloud:
    container_name: owncloud
    depends_on:
    - mariadb
    environment:
      OWNCLOUD_ADMIN_PASSWORD: redacted
      OWNCLOUD_ADMIN_USERNAME: admin
      OWNCLOUD_DB_HOST: mariadb
      OWNCLOUD_DB_NAME: owncloud
      OWNCLOUD_DB_PASSWORD: redacted
      OWNCLOUD_DB_TYPE: mysql
      OWNCLOUD_DB_USERNAME: owncloud
      OWNCLOUD_DOMAIN: yggdrasil
      OWNCLOUD_TRUSTED_DOMAINS: yggdrasil
      OWNCLOUD_MYSQL_UTF8MB4: 'true'
      OWNCLOUD_REDIS_ENABLED: 'false'
      OWNCLOUD_SUB_URL: /owncloud
    healthcheck:
      interval: 30s
      retries: 5
      test:
      - CMD
      - /usr/bin/healthcheck
      timeout: 10s
    hostname: owncloud
    image: owncloud/server:latest
    labels:
    - traefik.enable=true
    - traefik.http.services.owncloud.loadbalancer.server.port=8080
    - traefik.http.routers.owncloud.rule=PathPrefix(`/owncloud`)
    - traefik.http.routers.owncloud.entrypoints=web-ssl
    restart: unless-stopped
    volumes:
    - /media/data/owncloud-data:/mnt/data
    - /etc/ssl/certs/ca-certificates.crt:/var/www/owncloud/resources/config/ca-bundle.crt
```

config.php
```
<?php
$CONFIG = array (
  'apps_paths' =>
  array (
    0 =>
    array (
      'path' => '/var/www/owncloud/apps',
      'url' => '/apps',
      'writable' => false,
    ),
    1 =>
    array (
      'path' => '/var/www/owncloud/custom',
      'url' => '/custom',
      'writable' => true,
    ),
  ),
  'trusted_domains' =>
  array (
    0 => 'yggdrasil',
  ),
  'datadirectory' => '/mnt/data/files',
  'dbtype' => 'mysql',
  'dbhost' => 'mariadb:3306',
  'dbname' => 'owncloud',
  'dbuser' => 'owncloud',
  'dbpassword' => 'redacted',
  'dbtableprefix' => 'oc_',
  'log_type' => 'owncloud',
  'supportedDatabases' =>
  array (
    0 => 'sqlite',
    1 => 'mysql',
    2 => 'pgsql',
  ),
  'upgrade.disable-web' => true,
  'default_language' => 'en',
  'htaccess.RewriteBase' => '/owncloud',
  'logfile' => '/mnt/data/files/owncloud.log',
  'loglevel' => 2,
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'mysql.utf8mb4' => true,
  'filelocking.enabled' => true,
  'passwordsalt' => 'redacted',
  'secret' => 'redacted',
  'version' => '10.12.0.6',
  'logtimezone' => 'UTC',
  'installed' => true,
  'instanceid' => 'redacted',
  'maintenance' => false,
  'allow_user_to_change_mail_address' => '',
);
```

**Comment by xoxys** [comment]
Agree, looks like a core issue. Thanks @ho4ho!

**Comment by xoxys** [comment]
I'm going to close this by https://doc.owncloud.com/docs/next/server_release_notes.html#owncloud-inaccessibility A fix will be shipped in the next onwCloud 10.12.1 core release.
